CT2935, CT-2936, CT-2970 & CT-2981 Letter template amendments

njpreddy opened the pull request:
## Description
CT-2935 Prison number to appear in address for all Prisoner Acknowledgement & Disclosure letters.
        "[DELETE AS APPROPRIATE]" added when multiple prison numbers are detected.

CT-2936 Dispatch cover letter needs relevant department head.

CT-2970 Date of request appears in Acknowledgement letters

CT-2981 Prison number to appear in address for all Solicitor Acknowledgement & Disclosure letters.
        "[DELETE AS APPROPRIATE]" added when multiple prison numbers are detected.

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
<!-- Screenshots of the new changes if appropriate -->

### Related JIRA tickets
<!-- A link or list of links to relevant issues in Jira -->
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2935
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2936
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2970
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2981

### Deployment
<!-- Notes about database migrations, new runtime dependencies, mitigating downtime, feature flags, etc -->

### Manual testing instructions
<!-- Step-by-step instructions for the reviewer to manually test the changes -->
1) CT-2936 Dispatch cover letter needs relevant department head.
  Create an offender SAR request and progress it to 'dispatch' stage.
  Click "Send dispatch letter".
  Choose "Prisoner disclosed cover letter"
  The address on the html page and word document should contain place holders to be deleted as appropriate:

![CT-2936 prisoner disp cover letter](https://user-images.githubusercontent.com/6988369/89526637-f49f9780-d7df-11ea-8c44-ab96f839b810.png)

----------------------------------------------------------------

2) CT-2970 Date of request appears in prisoner, solicitor acknowledgement letters

![CT-2970 prisoner ack letter](https://user-images.githubusercontent.com/6988369/89526922-6841a480-d7e0-11ea-8f98-7de7bd39a4e4.png)

![CT-2970 solicitor ack letter](https://user-images.githubusercontent.com/6988369/89526941-6f68b280-d7e0-11ea-9e87-f3db05ecb1c5.png)

----------------------------------------------------------------

3) CT-2935 Prison number to appear in address for all Prisoner Acknowledgement & Disclosure letters.
   "[DELETE AS APPROPRIATE]" added when multiple prison numbers are detected.

   Prison number should appear in correspondence according to the following scenarios: 

<img width="558" alt="Acknowledgement correspondense table" src="https://user-images.githubusercontent.com/6988369/89527034-932bf880-d7e0-11ea-90bd-46c2d73c5275.png">

![CT-2935 prisoner ack letter](https://user-images.githubusercontent.com/6988369/89527168-d0908600-d7e0-11ea-9540-81667237377f.png)

![CT-2935 solicitor ack letter](https://user-images.githubusercontent.com/6988369/89527189-d71efd80-d7e0-11ea-8d2b-d34f4f7d86e1.png)

----------------------------------------------------------------

4) CT-2981 Prison number to appear in address for all Solicitor Acknowledgement & Disclosure letters.
   "[DELETE AS APPROPRIATE]" added when multiple prison numbers are detected.

   Prison number should appear in correspondence according to the following scenarios: 

<img width="558" alt="Dispatch correspondense table" src="https://user-images.githubusercontent.com/6988369/89527279-fb7ada00-d7e0-11ea-98ee-0bef9c980a6f.png">

![CT-2981 prisoner disp cover letter](https://user-images.githubusercontent.com/6988369/89527353-1a796c00-d7e1-11ea-83c2-13dcf792e89c.png)

![CT-2981 prisoner disp letter](https://user-images.githubusercontent.com/6988369/89527366-2107e380-d7e1-11ea-9d7d-032dfaa6b037.png)

![CT-2981 solicitor disp letter](https://user-images.githubusercontent.com/6988369/89527384-26fdc480-d7e1-11ea-9794-f7801e872aed.png)

![CT-2981 third party disp letter](https://user-images.githubusercontent.com/6988369/89527405-3250f000-d7e1-11ea-8f5b-d012db9edc14.png)

----------------------------------------------------------------

5) Additionally, only letters going to solicitors should be addressed "Dear Sirs".  This text should not appear in letters going to a third party.

njpreddy committed:
CT-2935 Prison number in inmate letters address

Add prison_numbers method to detect comma of space delimited list of
prison numbers. If found, add text [DELETE AS APPROPRIATE] to end of
list.

njpreddy committed:
CT-2935 Refactor prison number method

njpreddy committed:
CT-2935

* Increase rubocop ClassLength metric value.
* Letter template shows SAR requested date.
* Address details moved to letter template class.
* [DELETE AS APPROPRIATE] added to end of multiple prison number output.
* Amend docx letter template header text placed in single 2x2 table.
* Remove docx letter template addressee & address fields.

njpreddy committed:
CT-2935

-Add address placeholder in prison disclosed cover letter.
-Add "Dear Sirs" to solicitor letters where the requester = recipient.

njpreddy committed:
Merge branch 'master' into ct-2935-prison-number-in-address-for-inmate-letters

willmcb reviewed (commented):
See the comments - other than these small things this looks good 👍

Polsonby reviewed (changes requested):
One issue, please could we fix the strftime from blowing up if the received date is nil?

njpreddy committed:
Merge branch 'master' into ct-2935-prison-number-in-address-for-inmate-letters

njpreddy committed:
CT-2935

*Revert global Rubocop Metrics/ClassLength to 300 lines.
*Disable Rubocop Metrics/ClassLength check in ConfigValidator class.
*Apply Safe Navigation Operator (&.) to letter requested date fields.

njpreddy committed:
Merge local ct-2935 branch with origin ct-2935

njpreddy committed:
Merge branch 'master' into ct-2935-prison-number-in-address-for-inmate-letters

njpreddy committed:
CT-2935

*Revert breaking merge changes to structure.sql file.

njpreddy committed:
Merge branch 'master' into ct-2935-prison-number-in-address-for-inmate-letters

Commit:
Merge branch 'master' into ct-2935-prison-number-in-address-for-inmate-letters
